OADP-1791: Added suggestions for the backup types (#93556)
openshift/openshift-docs: backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc

You can back up applications by creating a `Backup` custom resource (CR). For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-creating-backup-cr.adoc#oadp-creating-backup-cr-doc[Creating a Backup CR].
The following are the different backup types for a `Backup` CR:

ifdef::openshift-rosa,openshift-rosa-hcp[]
The `Backup` CR creates backup files for Kubernetes resources and internal images on S3 object storage.
endif::openshift-rosa,openshift-rosa-hcp[]

ifndef::openshift-rosa,openshift-rosa-hcp[]
* The `Backup` CR creates backup files for Kubernetes resources and internal images on S3 object storage.

> * For an on-cluster backup, applications and persistent data are fully backed up. This type of backup is fast and can be executed quite often. But the data is vulnerable to cluster or storage outages. You can use CSI volume snapshots.

Reviewer: In an on-cluster backup, applications and persistent data are fully backed up within the cluster. This type of backup is fast and can be executed frequently. However, the data remains vulnerable to the cluster or storage outages. You can use CSI volume snapshots for this backup type.

> * For an on-cluster backup, applications and persistent data are fully backed up. This type of backup is fast and can be executed quite often. But the data is vulnerable to cluster or storage outages. You can use CSI volume snapshots.

Reviewer 2: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-semicolons#semicolons-between-independent-clauses

> * For an on-cluster backup, applications and persistent data are fully backed up. This type of backup is fast and can be executed quite often. But the data is vulnerable to cluster or storage outages. You can use CSI volume snapshots.

Reviewer 2: Expand CSI first occurence??

> * For an on-cluster backup, applications and persistent data is fully backed up. This type of backup is fast and can be executed quite often but the data is vulnerable to cluster or storage outages on the cluster. You can use CSI volume snapshots.

Reviewer 3: ```suggestion
* For an on-cluster backup, applications and persistent data are fully backed up. This type of backup is fast and can be executed quite often but the data is vulnerable to cluster or storage outages on the cluster. You can use CSI volume snapshots.
```

> * For an off-cluster backup, the Openshift object data and persistent data are backed up and moved from a cluster to ensure that a cluster outage does not impact the backup. This type of backup is slower as compared to a on-cluster backup, and should be executed less often than on-cluster. You can use CSI Data Mover backup and Kopia FSB.

Reviewer: In an off-cluster backup, the Openshift object data and persistent data are backed up and moved from the cluster to protect against cluster outages. This type of backup is slower than an on-cluster backup, and should be executed less frequently. You can use CSI Data Mover backup and Kopia FSB for off-cluster backup.

> * For an off-cluster backup, the Openshift object data and persistent data are backed up and moved from a cluster to ensure that a cluster outage does not impact the backup. This type of backup is slower as compared to a on-cluster backup, and should be executed less often than on-cluster. You can use CSI Data Mover backup and Kopia FSB.

Reviewer 2: ```suggestion
* In an off-cluster backup, you back up the Openshift object data and persistent data and move them from a cluster to ensure that a cluster outage does not impact the backup. This type of backup is slower than an on-cluster backup; and therefore you should execute it less frequently than an on-cluster backup. You can use CSI Data Mover backup and Kopia FSB for an off-cluster backup.
```

Reviewer 2: Kopia FSB > Is there an expansion?? if so, you need to use it for the first occurence?

> * Executing a single backup that performs both on-cluster and off-cluster backup is not possible.

Reviewer 2: ```suggestion
* You can only execute a single backup that performs either an on-cluster or off-cluster backup.
```

Reviewer 2: https://www.ibm.com/docs/en/ibm-style?topic=medium-global-audiences#style

Avoid negative sentence constructions.

> * In an off-cluster backup, you back up the Openshift object data and persistent data and move them from a cluster to ensure that a cluster outage does not impact the backup. This type of backup is slower than an on-cluster backup; and therefore you should execute it less frequently than an on-cluster backup. You can use CSI Data Mover backup and Kopia File System Backup (FSB) for an off-cluster backup.

Reviewer 4: I think it should be whether you would like to store persistent volume data in bucket or not.  Yamls are always stored in bucket irrespective of type of backup

You can back up applications by creating a `Backup` custom resource (CR). For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-creating-backup-cr.adoc#oadp-creating-backup-cr-doc[Creating a Backup CR].
The following are the different backup types for a `Backup` CR:

ifdef::openshift-rosa,openshift-rosa-hcp[]
The `Backup` CR creates backup files for Kubernetes resources and internal images on S3 object storage.
endif::openshift-rosa,openshift-rosa-hcp[]

ifndef::openshift-rosa,openshift-rosa-hcp[]
* The `Backup` CR creates backup files for Kubernetes resources and internal images on S3 object storage.

> * In an on-cluster backup, you back up the persistent volume data within the cluster. This type of backup is fast, and you can execute it frequently; however, the data is vulnerable to cluster or storage outages. You can use Container Storage Interface (CSI) volume snapshots for an on-cluster backup.

Reviewer 4: @weshayutin @apurvabhide17 I'm thinking change it to something like this 
```
if you use velero's snapshot feature to backup volumes in that case only snapshot related information is stored in s3 bucket along with the openshift object data. If undelying storage is part of same cluster infra then data might be loss in case of disaster.  Same for bucket if bucket is part of the same cluster then all backup data will be lost in case of disaster.
```
